database_models.go

```diff
@@ -112,18 +112,19 @@ var buildJSONToColumns = map[string]string{
 // Build holds data about the state of a build. Which parameters was used to
 // start it, what status it holds, et.al.
 type Build struct {
-	BuildID     uint         `gorm:"primaryKey" json:"buildId"`
-	StatusID    BuildStatus  `gorm:"not null" json:"statusId"`
-	ProjectID   uint         `gorm:"not null;index:build_idx_project_id" json:"projectId"`
-	Project     *Project     `gorm:"foreignKey:ProjectID;constraint:OnUpdate:RESTRICT,OnDelete:RESTRICT" json:"-"`
-	ScheduledOn *time.Time   `gorm:"nullable;default:NULL" json:"scheduledOn" format:"date-time"`
-	StartedOn   *time.Time   `gorm:"nullable;default:NULL" json:"startedOn" format:"date-time"`
-	CompletedOn *time.Time   `gorm:"nullable;default:NULL" json:"finishedOn" format:"date-time"`
-	GitBranch   string       `gorm:"size:300;default:'';not null" json:"gitBranch"`
-	Environment null.String  `gorm:"nullable;size:40" json:"environment" swaggertype:"string"`
-	Stage       string       `gorm:"size:40;default:'';not null" json:"stage"`
-	Params      []BuildParam `gorm:"foreignKey:BuildID" json:"params"`
-	IsInvalid   bool         `gorm:"not null;default:false" json:"isInvalid"`
+	BuildID             uint                `gorm:"primaryKey" json:"buildId"`
+	StatusID            BuildStatus         `gorm:"not null" json:"statusId"`
+	ProjectID           uint                `gorm:"not null;index:build_idx_project_id" json:"projectId"`
+	Project             *Project            `gorm:"foreignKey:ProjectID;constraint:OnUpdate:RESTRICT,OnDelete:RESTRICT" json:"-"`
+	ScheduledOn         null.Time           `gorm:"nullable;default:NULL" json:"scheduledOn" format:"date-time"`
+	StartedOn           null.Time           `gorm:"nullable;default:NULL" json:"startedOn" format:"date-time"`
+	CompletedOn         null.Time           `gorm:"nullable;default:NULL" json:"finishedOn" format:"date-time"`
+	GitBranch           string              `gorm:"size:300;default:'';not null" json:"gitBranch"`
+	Environment         null.String         `gorm:"nullable;size:40" json:"environment" swaggertype:"string"`
+	Stage               string              `gorm:"size:40;default:'';not null" json:"stage"`
+	Params              []BuildParam        `gorm:"foreignKey:BuildID" json:"params"`
+	IsInvalid           bool                `gorm:"not null;default:false" json:"isInvalid"`
+	TestResultSummaries []TestResultSummary `gorm:"foreignKey:BuildID" json:"testResultSummaries"`
```

Comment:
All spacing here is changed resulting in a 100% diff for these lines. Ideally the spacing change would be in different PR to make this more reviewable.

There is a hidden `null.Time` -> `*time.Time ` in this change- Intentional?

Reply:
The change from `*time.Time` to using `null.Time` was intentional, more of a drive-by change than required for the implementation.


I agree that it really makes it harder to see what's going on. I think it would cause more trouble if we tried to avoid it though, as it gets formatted like that when you run `goimports -w .`, and the pros of formatting while coding outweigh the cons in this case IMO. Especially if it would mean an extra "formatting" PR after basically every PR that touches a struct.

This is mainly an issue for structs I believe, and big ones at that, so I think while it's not great when it's something like this, it shouldn't happen very often. One solution/workaround could be to use `git diff --word-diff` if you're checking locally, or https://diffonline.net/ [4-color (light/dark red/green) highlighting] or https://text-compare.com/ [1-color highlighting].

Unless there is a way to change the GitHub diff settings somehow, which would be great.

Reply:
There is a way to hide changes if it's only whitespace changes on the given line.

![Screenshot from 2021-09-15 08-01-02](https://user-images.githubusercontent.com/2477952/133379101-4feed10e-2cc9-41a8-98d8-2d2c7b6cc551.png)

Before:

![Screenshot from 2021-09-15 08-02-18](https://user-images.githubusercontent.com/2477952/133379199-4813463c-83fe-4497-b45e-ca984e8fe11e.png)

After:

![Screenshot from 2021-09-15 08-00-27](https://user-images.githubusercontent.com/2477952/133379193-8920ca65-2268-44d2-84fa-a826a413a1a3.png)

Reply:
Awesome! 👍

Reply:
@fredx30 Mentioning to make sure you see it too 👍🏻